src/main/scala/com/fulcrumgenomics/bam/CallConsensusFromUmis.scala

```diff
@@ -0,0 +1,118 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.bam.ConsensusCallerOptions._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.{PhredValue, ProgressLogger}
+import dagr.commons.CommonsDef.PathToBam
+import dagr.commons.io.Io
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import dagr.sopt.cmdline.ValidationException
+import htsjdk.samtools._
+import htsjdk.samtools.util.CloserUtil
+
+import scala.collection.JavaConverters._
+
+
+@clp(description =
+  """
+    |Calls consensus sequences from reads with the same UMI identifier.
+    |
+    |This tool assumes that reads with the same identifier are grouped together (consecutive in the file).
+  """,
+  group = ClpGroups.SamOrBam)
+class CallConsensusFromUmis
+(  @arg(flag="i", doc="The input SAM or BAM file.") val input: PathToBam,
+   @arg(flag="o", doc="Output SAM or BAM file to write consensus reads.") val output: PathToBam,
+   @arg(flag="r", doc="Output SAM or BAM file to write reads not used.") val rejects: PathToBam,
+   @arg(flag="t", doc="The SAM attribute with the unique UMI identifier.") val attribute: String = DefaultAttribute,
+   @arg(flag="p", doc="The Prefix all consensus read names") val readNamePrefix: Option[String] = None,
+   @arg(flag="R", doc="The Read Group for all the consensus reads.") val readGroup: String = "A",
+   @arg(flag="1", doc="The Phred-scaled error rate for an error prior to the UMIs being integrated.") val errorRatePreUmi: PhredValue = DefaultErrorRatePreUmi,
+   @arg(flag="2", doc="The Phred-scaled error rate for an error post the UMIs have been integrated.") val errorRatePostUmi: PhredValue = DefaultErrorRatePostUmi,
+   @arg(flag="q", doc="Cap the maximum base quality in the input.") val maxBaseQuality: PhredValue = DefaultMaxBaseQuality,
+   @arg(flag="s", doc="Subtract this base quality from the input base qualities.") val baseQualityShift: PhredValue = DefaultBaseQualityShift,
+   @arg(flag="N", doc="Mask (make 'N') consensus bases with quality less than this threshold.") val minConsensusBaseQuality: PhredValue = DefaultMinConsensusBaseQuality,
+   @arg(flag="M", doc="The minimum number of reads to produce a consensus base.") val minReads: Int = DefaultMinReads,
+   @arg(flag="Q", doc="The minimum mean base quality across a consensus base to output.") val minMeanConsensusBaseQuality: PhredValue = DefaultMinMeanConsensusBaseQuality
+  ) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  Seq(output, rejects).foreach(Io.assertCanWriteFile(_))
+  if (attribute.length != 2) throw new ValidationException("attribute must be of length 2")
+  if (errorRatePreUmi < 0)   throw new ValidationException("Error rate pre UMI must be >= 0")
```

Comment:
Yes, `0` will be converted to `PositiveInfinity` as it should be, thus zero'ing out the cases where there is an error pre or post UMI (not in sequencing).  The conversion is implicit for now, as I am likely going to remove those implicits since they are bug prone.